### PR TITLE
Fix 403 error on POST and PUT requests when logged in

### PR DESCRIPTION
**Pull request by ryan-conn**
## Description
Fixes CSRF (403) errors when the user is logged in

## Rationale

After logging in, DRF uses `SessionAuthentication` to process requests, which needs the `X-CSRFToken` cookie to be set (see [this page from the DRF docs](https://www.django-rest-framework.org/topics/ajax-csrf-cors/)). This PR adds this to our PUT/POST request headers.

I didn't make any tests for this because I don't think it's possible/feasible to test how our frontend and backend interact and couldn't think of any other way to test that it works, but let me know if anyone has any ideas.

## How to test

Log in, and now all of our API routes work regardless of whether you're logged in

## Related tasks

We never made an issue for this, but everyone's run into the 403 errors at some point

### CODE REVIEW

**Review by gannonprudhomme — APPROVED** [review]

File: autoscheduler/frontend/src/package.json [hunk 1]
```diff
@@ -58,10 +58,12 @@
     "@material-ui/styles": "^4.9.10",
     "@reach/router": "^1.2.1",
     "@teamsupercell/typings-for-css-modules-loader": "^2.1.0",
+    "@types/js-cookie": "^2.2.6",
```
[the remark this answers is not in the document]
Comment (ryan-conn) [reply autoscheduler/frontend/src/package.json:61]: Yeah I was also thinking the same thing when I put it there tbh, should I just move all of the `@types` packages into `devDependencies` in this PR? I moved them there on my local branch and ran linting + testing + the site without any issues, so I don't see any reason to keep them in `dependencies`.

[the remark this answers is not in the document]
Comment (gannonprudhomme) [reply autoscheduler/frontend/src/package.json:61]: Yeah I think so. Do you have any input @firejake308 ?

[the remark this answers is not in the document]
Comment (gannonprudhomme) [reply autoscheduler/frontend/src/package.json:61]: Regardless, I just say merge this and we do it in a different PR - don't think adding one is doing any more harm

[the remark this answers is not in the document]
Comment (firejake308) [reply autoscheduler/frontend/src/package.json:61]: Sure, we should definitely fix that, but I think it's best done in a separate PR

**Pull request merged**